src/live_restore/live_restore_private.h

```diff
@@ -55,8 +55,43 @@ struct __wt_live_restore_file_handle {
     } destination;
 
     WT_FS_OPEN_FILE_TYPE file_type;
+    WT_RWLOCK ext_lock; /* File extent list lock */
+
+/* AUTOMATIC FLAG VALUE GENERATION START 0 */
+#define WT_LIVE_RESTORE_FH_EXTENT_LIST_LOCKED 0x1u
+    /* AUTOMATIC FLAG VALUE GENERATION STOP 8 */
+    uint8_t flags;
 };
 
+/*
+ * WT_WITH_LIVE_RESTORE_EXTENT_LIST_READ_LOCK --
+ *     Acquire the extent list read lock and perform an operation.
+ */
+#define WT_WITH_LIVE_RESTORE_EXTENT_LIST_READ_LOCK(session, lr_fh, op) \
+    do {                                                               \
+        __wt_readlock((session), &(lr_fh)->ext_lock);                  \
+        F_SET((lr_fh), WT_LIVE_RESTORE_FH_EXTENT_LIST_LOCKED);         \
+        op;                                                            \
+        F_CLR((lr_fh), WT_LIVE_RESTORE_FH_EXTENT_LIST_LOCKED);         \
```

Comment:
This flag won't be correct for multiple readers. If we have two readers in the critical section and one leaves the flag will be cleared while the lock is still in use.

Do we need this flag? I'm not sure what benefit it has outside of debugging and we can always derive it from `l.u.s.readers_active`.

Reply:
I agree, this flag feels overkill and you are correct that the F_CLR means it's kinda meaningless in a multithreaded (readlock) context. @sueloverso I think we should drop it, what do you think?

Reply:
I have pro-actively removed the flag as I agree that it has no value really. I can revert the removing commit pending Sue's feedback. e69b28b54

Reply:
Yes, because we're setting it in a shared structure it is not effective. The other locks use a flag in the session.